### PR TITLE
test/e2e/framework: remove skip.go and use e2eskipper subpackage

### PR DESCRIPTION
**Pull request by tanjunchen**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

close:https://github.com/kubernetes/kubernetes/issues/87047

**Special notes for your reviewer**:

/assign @oomichi 

/hold
for waiting  https://github.com/kubernetes/kubernetes/pull/87270 and https://github.com/kubernetes/kubernetes/pull/87317 merged

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

**Comment by tanjunchen** [comment]
ref:https://github.com/kubernetes/kubernetes/issues/86763
/priority backlog

File: test/e2e/framework/skip.go [hunk 1]
```diff
@@ -1,185 +0,0 @@
...
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	utilversion "k8s.io/apimachinery/pkg/util/version"
-	utilfeature "k8s.io/apiserver/pkg/util/feature"
-	"k8s.io/client-go/discovery"
-	"k8s.io/client-go/dynamic"
-	clientset "k8s.io/client-go/kubernetes"
-	"k8s.io/kubernetes/pkg/features"
-
-	// TODO: Remove the following imports (ref: https://github.com/kubernetes/kubernetes/issues/81245)
```
Comment (tanjunchen) [inline test/e2e/framework/skip.go]: remove todo

**Review by fejta — COMMENTED** [review]
/approve
/lgtm

**Comment by tanjunchen** [comment]
/cc @oomichi 
test/e2e/framework/util.go  use the skipInternalf of skip.go , so can we remove this skip.go?

**Comment by oomichi** [comment]
The reason why the above jobs are failed is 
```
/b/f/w/test/e2e_node/topology_manager_test.go:235:3: undefined: "k8s.io/kubernetes/test/e2e/framework".Skipf
compilepkg: error running subcommand: exit status 2
```
So we need to make `test/e2e_node/topology_manager_test.go` switch to use e2eskipper`s one.

File: test/e2e/framework/util.go [hunk 2]
```diff
@@ -203,6 +204,12 @@ func GetMasterHost() string {
 	return masterURL.Hostname()
 }
 
+func skipInternalf(caller int, format string, args ...interface{}) {
```
Comment (tanjunchen) [inline test/e2e/framework/util.go]: now we need to keep this function in this file

File: test/integration/scheduler/framework_test.go [hunk 3]
```diff
@@ -872,9 +871,7 @@ func TestBindPlugin(t *testing.T) {
 			Enabled: []schedulerconfig.Plugin{{Name: unreservePlugin.Name()}},
 		},
 		Bind: &schedulerconfig.PluginSet{
-			// Put DefaultBinder last.
-			Enabled:  []schedulerconfig.Plugin{{Name: bindPlugin1.Name()}, {Name: bindPlugin2.Name()}, {Name: defaultbinder.Name}},
-			Disabled: []schedulerconfig.Plugin{{Name: defaultbinder.Name}},
```
Comment (oomichi) [inline test/integration/scheduler/framework_test.go]: Why do we need to change here?

File: test/e2e/framework/util.go (hunk 2 above)
Comment (oomichi) [reply test/e2e/framework/util.go]: It is better to write the reason why as `// NOTE` or something around this function.

**Review by oomichi — COMMENTED** [review]
Thanks for doing this, just few comments

File: test/e2e/framework/util.go (hunk 2 above)
Comment (alejandrox1) [reply test/e2e/framework/util.go]: I see that this function is used within the framework/util.go at `RunIfContainerRuntimeIs` and `RunIfSystemSpecNameIs`.
What do you all think about moving these 2 functions to the e2eskipper pkg as well?
The name of these functions doesn't start with "skip" but other than that it has a similar behaviour.
And with those functions we can also move this out of the framework.

**Review by alejandrox1 — COMMENTED** [review]
/milestone v1.18

**Comment by oomichi** [comment]
@tanjunchen Can you remove `/hold` from this PR message?
In addition, `ref:#87047` can be replaced with `close: #87047` because this PR makes the issue finish.

One more thing, can you check the above comments you might not see yet?

**Comment by tanjunchen** [comment]
@oomichi sure 
i will update it right now.Thanks

File: test/e2e/framework/util.go (hunk 2 above)
Comment (tanjunchen) [reply test/e2e/framework/util.go]: good idea

File: test/integration/scheduler/framework_test.go (hunk 3 above)
Comment (tanjunchen) [reply test/integration/scheduler/framework_test.go]: sorry, udpate it

File: test/integration/scheduler/framework_test.go [hunk 4]
```diff
@@ -28,7 +28,7 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/wait"
 	clientset "k8s.io/client-go/kubernetes"
-	scheduler "k8s.io/kubernetes/pkg/scheduler"
+	"k8s.io/kubernetes/pkg/scheduler"
```
Comment (oomichi) [inline test/integration/scheduler/framework_test.go]: Please don't change unrelated file.

Comment (tanjunchen) [reply test/integration/scheduler/framework_test.go]: sure .
i will update it

**Comment by oomichi** [comment]
Thanks for doing this.
This looks nice and clean.

/lgtm
